os_net_config/impl_ifcfg.py

```
@@ -1984,6 +1987,11 @@ def apply(self, cleanup=False, activate=True, config_rules_dns=True):
                         )
                         self.ifdown(interface_name)
                         self.remove_config(ifcfg_file)
+            for nmconn_file in glob.iglob(nm_cleanup_pattern()):
+                interface_name = Path(nmconn_file).stem
+                cmd = ['nmcli', 'connection', 'delete', interface_name]
+                self.execute(msg, *cmd)
```

It will be good to keep this self.execute inside of a try except block so that in case of failure its doesnt silently fail.

---

Not sure what you mean by fail silently. It would be the same as ifdown() call in L1985 which also uses cli command. Unless you want to log something here or ignore the error I don't see why one would use try/except block.